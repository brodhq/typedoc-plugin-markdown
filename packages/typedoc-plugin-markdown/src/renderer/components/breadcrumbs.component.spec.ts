import { TestApp } from '../../../test/test-app';
import { BreadcrumbsComponent } from './breadcrumbs.component';

describe(`Breadcrumbs Component`, () => {
  let testApp: TestApp;

  beforeAll(async () => {
    testApp = new TestApp(['classes.ts', 'variables.ts']);
    await testApp.bootstrap();
  });

  test(`should compile breadcrumbs for a entrypoint'`, () => {
    const page = testApp.getPageByUrl('modules.md').model;
    expect(BreadcrumbsComponent(page)).toMatchSnapshot();
  });

  test(`should compile breadcrumbs for a module'`, () => {
    const page = testApp.getPageByModelName('classes').model;
    expect(BreadcrumbsComponent(page)).toMatchSnapshot();
  });

  test(`should compile breadcrumbs for a reflection'`, () => {
    const page = testApp.getPageByModelName('Point').model;
    expect(BreadcrumbsComponent(page)).toMatchSnapshot();
  });
});
