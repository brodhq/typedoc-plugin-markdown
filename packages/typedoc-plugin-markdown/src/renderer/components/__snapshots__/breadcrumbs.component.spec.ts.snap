// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs Component should compile breadcrumbs for a entrypoint' 1`] = `"Modules"`;

exports[`Breadcrumbs Component should compile breadcrumbs for a module' 1`] = `"[Modules](../modules.md) / classes"`;

exports[`Breadcrumbs Component should compile breadcrumbs for a reflection' 1`] = `"[Modules](../modules.md) / [classes](../modules/classes.md) / Point"`;
